src/memory/memory_shared.h

```diff
@@ -90,12 +91,33 @@ typedef union {
         uint8_t ble_bond_db[MEMORY_BLE_BOND_DB_LEN];
         uint8_t ble_identity_resolving_key[MEMORY_BLE_IRK_LEN];
         uint8_t ble_identity_address[MEMORY_BLE_ADDR_LEN];
-        uint8_t padding[2]; // align to 4 bytes
+        uint8_t reserve2[2]; // align to 4 bytes
+        // Hash of the BLE firmware that is allowed to be loaded into the BLE chip.
+        uint8_t ble_allowed_firmware_hash[32];
+        // - 0xFF: uninitialized (should never happen after factorysetup)
+        // - 0x00: active BLE firmware is at `MEMORY_SPI_BLE_FIRMWARE_1_ADDR` in the SPI memory
+        // chip.
+        // - 0x01: active BLE firmware is at `MEMORY_SPI_BLE_FIRMWARE_2_ADDR` in the SPI memory
+        // chip.
+        uint8_t ble_active_index;
+        // Checksum of each of the two BLE stored firmwares.
+        //
+        // The checksum is the xor of all bytes. It is returned by the BLE chip after loading the
+        // firmware, so we can verify that the firmware was loaded successfully.
+        uint8_t ble_firmware_checksums[2];
+        uint8_t reserve3[1]; // align to 4 bytes
+        // Size of each of the two stored BLE firmwares.
+        uint16_t ble_firmware_sizes[2];
+        uint8_t reserve4[2]; // align to 4 bytes
```

Comment:
is reserve4 needed, 2 uint16s are 4 bytes.

Reply:
Correct :heavy_check_mark:  Removed